DevelopGitHub/GitHubExtension/GitHubExtension.LocalizationTool/Translate/Translator.cs

```diff
@@ -0,0 +1,186 @@
+                    return "uk-UK";
+                case Lang.Us:
+                    return "en-US";
+                case Lang.Ru:
+                    return "ru-RU";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public static string GetFileName(Lang lang)
+        {
+            switch (lang)
+            {
+                case Lang.Uk:
+                    return "uk-UK.json";
+                case Lang.Us:
+                    return "en-US.json";
+                case Lang.Ru:
+                    return "ru-RU.json";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public void AddNewRowToTranslationData(Lang language, KeyValuePair<string, JToken> element)
+        {
+            var line = new TranslationDataRow(element.Key);
+            line[language] = element.Value.ToString();
+            TranslationData.Add(line);
+        }
+
+        public async void WebTranslate(string sourceLanguage, string targetLanguage)
+        {
+            if (sourceLanguage.Equals(targetLanguage))
+            {
+                return;
+            }
+
+            Lang sourceLanguageEnum;
+            Lang targetLanguageEnum;
+            TargetLanguageEnum(ref sourceLanguage, ref targetLanguage, out sourceLanguageEnum, out targetLanguageEnum);
+
+            StringBuilder textToTranslate = GenerateTextParameter(sourceLanguageEnum);
+
+            var result = Task<JObject>.Factory.StartNew(() => GetTranslationFromYandexApi(sourceLanguage, targetLanguage, textToTranslate));
+            mainWindow.TranslateButton.IsEnabled = false;
+            mainWindow.TranslateButton.Content = "Translating...";
+            await result;
+            mainWindow.TranslateButton.IsEnabled = true;
+            mainWindow.TranslateButton.Content = "WebTranslate";
+
+            if (result.Result == null)
+            {
+                return;
+            }
+
+            SaveTranslationResult(result, targetLanguageEnum);
+        }
+
+        public void RemoveEmptyDataRows()
+        {
+            for (var i = 0; i < TranslationData.Count; i++)
+            {
+                if (string.IsNullOrWhiteSpace(TranslationData[i].Name))
+                {
+                    TranslationData.Remove(TranslationData[i--]);
+                }
+            }
+        }
+
+        private static void TargetLanguageEnum(ref string sourceLanguage, ref string targetLanguage, out Lang sourceLanguageEnum, out Lang targetLanguageEnum)
+        {
+            sourceLanguageEnum = GetLang(sourceLanguage);
+            targetLanguageEnum = GetLang(targetLanguage);
+            sourceLanguage = sourceLanguage == "us" ? "en" : sourceLanguage;
+            targetLanguage = targetLanguage == "us" ? "en" : targetLanguage;
+        }
+
+        private static JObject GetTranslationFromYandexApi(
```

Comment:
This method returns only response (string -> JObject) of my request.
Parsing of this result takes place in other method (SaveTranslationResult)